aptos-move/framework/supra-framework/sources/automation_registry_state.move

```diff
@@ -0,0 +1,623 @@
+        /// Automation task index in registry
+        id: u64,
+        /// The address of the task owner.
+        owner: address,
+        /// The function signature associated with the registry entry.
+        payload_tx: vector<u8>,
+        /// Expiry of the task, represented in a timestamp in second.
+        expiry_time: u64,
+        /// The transaction hash of the request transaction.
+        tx_hash: vector<u8>,
+        /// Max gas amount of automation task
+        max_gas_amount: u64,
+        /// Maximum gas price cap for the task
+        gas_price_cap: u64,
+        /// Registration epoch number
+        registration_epoch: u64,
+        /// Registration epoch time
+        registration_time: u64,
+        /// Flag indicating whether the task is active, canclled or pending.
+        state: u8
+    }
+
+    #[event]
+    /// Update automation gas limit event
+    struct UpdateAutomationGasLimit has drop, store {
+        automation_gas_limit: u64
+    }
+
+    #[event]
+    /// Remove automation task registry event
+    struct CanclledAutomationTask has drop, store {
+        id: u64
+    }
+
+    public fun task_expiry_time(task: &AutomationTaskMetaData): u64 {
+        task.expiry_time
+    }
+
+    // todo : this function should call during initialzation, but since we already done genesis in that case who can access the function
+    public(friend) fun initialize(supra_framework: &signer, automation_gas_limit: u64) {
+        system_addresses::assert_supra_framework(supra_framework);
+
+        move_to(supra_framework, AutomationRegistryState {
+            tasks: enumerable_map::new_map(),
+            current_index: 0,
+            gas_committed_for_next_epoch: 0,
+            automation_gas_limit
+        })
+    }
+    public(friend) fun on_new_epoch(epoch_interval_micro: u64) acquires AutomationRegistryState {
+        let state = borrow_global_mut<AutomationRegistryState>(@supra_framework);
+        let ids = enumerable_map::get_map_list(&state.tasks);
+
+        let epoch_interval_secs = epoch_interval_micro / MICROSECS_CONVERSION_FACTOR;
+        let current_time = timestamp::now_seconds();
+        let gas_committed_for_next_epoch = 0;
+
+        // Perform clean up and updation of state
+        vector::for_each(ids, |id| {
+            let task = enumerable_map::get_value_mut(&mut state.tasks, id);
+
+            // Tasks that are active during next epoch and are not cancled
+            if (task.state != CANCELLED && task.expiry_time > (current_time + epoch_interval_secs) ) {
+                gas_committed_for_next_epoch = gas_committed_for_next_epoch + task.max_gas_amount;
+            };
+
+            // Drop or activate task
+            if (task.expiry_time <= current_time || task.state == CANCELLED) {
+                enumerable_map::remove_value(&mut state.tasks, id);
+            } else {
+                task.state = ACTIVE;
+            }
+        });
```

Comment:
```suggestion
            if (task.state != CANCELLED && task.expiry_time > (current_time + epoch_interval_secs) ) {
                gas_committed_for_next_epoch = gas_committed_for_next_epoch + task.max_gas_amount;
                 task.state = ACTIVE;
            }

            else { 
                enumerable_map::remove_value(&mut state.tasks, id);
            }; 
               
            
        });
```

Reply:
This function is called when a chain enters a new epoch, and current time will show the start of the new epoch.
with the 1st `if` statement we are collecting committed-gas-amount for the next epoch already.
With the second `if` statement we are clearing up already expired or canceled tasks for this current epoch.
So with the suggested change above we will remove also the tasks that are valid for the the current epoch but expired for the next upcoming epoch.